test/dynamo/test_triton_kernels.py

```diff
@@ -825,6 +825,34 @@ def f(inp):
         compiled_out = torch.compile(f)(inp)
         self.assertEqual(compiled_out, eager_out)
 
+    @requires_cuda()
+    def test_triton_kernel_slice_and_view_input(self):
```

Comment:
Before your change, what error would this test case produce?

Reply:
Great question. With this line present:

https://github.com/pytorch/pytorch/blob/62496ffd0dd620e414ffda71a09389c30c87e2d7/torch/_higher_order_ops/triton_kernel_wrap.py#L237

`mark_node_as_mutating` here fails:

https://github.com/pytorch/pytorch/blob/62496ffd0dd620e414ffda71a09389c30c87e2d7/torch/_inductor/ir.py#L4076-L4078

because non-RV views don't have the `layout` attribute. It makes this far, because the line above drops the `in_ptr` form the `tensors_to_clone` (because the `double_strided_kernel` is simple enough to be covered by the current mutation tracker), so we never get to cloning.

But if I comment out the `filter_non_mutated` line above, I get almost 100% numeric mismatch. This is because cloning then happens and the strides are not preserved (because the "preserve RVs" logic doesn't apply to non-RV views").

After this PR, the test passes both with and without the `filter_non_mutated` line.